content/en/docs/tasks/administer-cluster/encrypt-data.md

```diff
@@ -65,72 +120,160 @@ resources:
+By default, the `identity` provider is used to protect data in etcd, which provides **no**
+encryption. `EncryptionConfiguration` was introduced to encrypt confidential data locally, using
+a locally managed key.
 
 Encrypting secret data with a locally managed key protects against an etcd compromise, but it fails to
-protect against a host compromise. Since the encryption keys are stored on the host in the
-EncryptionConfiguration YAML file, a skilled attacker can access that file and extract the encryption
-keys.
+protect against a host compromise. Since the encryption keys are readable - in the clear - on the
+control plane host in the EncryptionConfiguration YAML file, a skilled attacker can access that file
+and extract the encryption keys.
 
 Envelope encryption creates dependence on a separate key, not stored in Kubernetes. In this case,
 an attacker would need to compromise etcd, the `kubeapi-server`, and the third-party KMS provider to
 retrieve the plaintext values, providing a higher level of security than locally stored encryption keys.
 
 ## Encrypting your data
 
-Create a new encryption config file:
+{{< caution >}}
+Storing the raw encryption key in the EncryptionConfig only moderately improves your security posture,
+compared to no encryption.
+
+For additional secrecy, consider using the `kms` provider as this relies on keys held outside your
+Kubernetes cluster. Implementations of `kms` can work with hardware security modules or with
+encryption services managed by your cloud provider.
+{{< /caution >}}
+
+These instructions explain how to set up non-KMS encryption. You can use these as a guide
+to how you would configure KMS encryption; however, the specific details of that depend
+on the (external) KMS integration that you are using.
+
+### Generate the encryption key
+
+Start by generating a new encryption key, and then encode it using base64:
+
+{{< tabs name="generate_encryption_key" >}}
+{{% tab name="Linux" %}}
+Generate a 32-byte random key and base64 encode it. You can use this command:
+```shell
+head -c 32 /dev/urandom | base64
+```
+
+You can use `/dev/hwrng` instead of `/dev/urandom` if you want to
+use your PC's built-in hardware entropy source. Not all Linux
+devices provide a hardware random generator.
+{{% /tab %}}
+{{% tab name="macOS" %}}
+<!-- localization note: this is similar to the Linux tab and the wording
+should match wherever the English text does -->
+Generate a 32-byte random key and base64 encode it. You can use this command:
+```shell
+head -c 32 /dev/urandom | base64
+```
+{{% /tab %}}
+{{% tab name="Windows" %}}
+Generate a 32-byte random key and base64 encode it. You can use this command:
+```powershell
+# Do not run this in a session where you have set a random number
+# generator seed.
+[Convert]::ToBase64String((1..32|%{[byte](Get-Random -Max 256)}))
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+
+{{< note >}}
+Keep the encryption secret confidential, including whilst you generate it and
```

Comment:
nit: Keep the encryption secret confidential, including while ...

Reply:
Let's call it a key, for consistency.
In this page, I'm always thinking of 'secret' as the thing to be encrypted.